src/cargo/ops/cargo_add/mod.rs

```diff
@@ -966,16 +966,53 @@ fn print_dep_table_msg(shell: &mut Shell, dep: &DependencyUI) -> CargoResult<()>
         } else {
             "".to_owned()
         };
+
         shell.write_stderr(format_args!("{}Features{}:\n", prefix, suffix), &style::NOP)?;
+
+        const MAX_FEATURE_PRINTS: usize = 50;
+
+        let mut activated_printed: usize = 0;
```

Comment:
nit: Rust compiler could infer the type for you :)

(ditto for deactivated)

```suggestion
        let mut activated_printed = 0;
```

Reply:
Old habits, removed.